src/vs/workbench/browser/parts/quickopen/quickOpenController.ts

```diff
@@ -549,6 +568,33 @@ export class QuickOpenController extends WorkbenchComponent implements IQuickOpe
 		return promiseCompletedOnHide;
 	}
 
+	private handleOnCancel(isPicker: boolean): void {
+		// restore the editor part state after cancelling
+		this.historyService.block(true);
```

Comment:
Good point, the goto line handler was causing the history to add stuff when previewing. I think also the gotoSymbolHandler.ts does! For those you can check https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/search/browser/openFileHandler.ts#L119 how we pass in the instantiation service and use that to create the entries. Doing so, you can get hold of any service via the constructor.